Sources/Interaction/Style/InteractorStyleManipulator/index.js

```diff
@@ -179,74 +179,68 @@ function vtkInteractorStyleManipulator(publicAPI, model) {
   };
 
   //-------------------------------------------------------------------------
-  publicAPI.removeMouseManipulator = (index) => {
-    if (model.mouseManipulators.length > index) {
-      model.mouseManipulators[index] = null;
+  const removeManipulator = (manipulator, list) => {
+    const index = list.indexOf(manipulator);
+    if (index !== -1) {
+      return false;
     }
+    list.slice(index, 1);
+    publicAPI.modified();
+    return true;
   };
 
   //-------------------------------------------------------------------------
-  publicAPI.removeVRManipulator = (index) => {
-    if (model.vrManipulators.length > index) {
-      model.vrManipulators[index] = null;
-    }
-  };
+  publicAPI.removeMouseManipulator = (manipulator) =>
+    removeManipulator(manipulator, model.mouseManipulators);
 
   //-------------------------------------------------------------------------
-  publicAPI.removeGestureManipulator = (index) => {
-    if (model.gestureManipulators.length > index) {
-      model.gestureManipulators[index] = null;
-    }
-  };
+  publicAPI.removeVRManipulator = (manipulator) =>
+    removeManipulator(manipulator, model.vrManipulators);
 
   //-------------------------------------------------------------------------
-  publicAPI.getMouseManipulator = (index) => {
-    let manipulator = null;
-    if (model.mouseManipulators.length > index) {
-      manipulator = model.mouseManipulators[index];
-    }
-    return manipulator;
-  };
+  publicAPI.removeGestureManipulator = (manipulator) =>
+    removeManipulator(manipulator, model.gestureManipulators);
 
   //-------------------------------------------------------------------------
-  publicAPI.getVRManipulator = (index) => {
-    let manipulator = null;
-    if (model.vrManipulators.length > index) {
-      manipulator = model.vrManipulators[index];
-    }
-    return manipulator;
+  const getManipulator = (index, list) => {
+    return index > list.length ? null : list[index];
```

Comment:
if index is out of bound, it would be fine. You should just do 

return list[index];

assuming we want to keep those getters?

Reply:
We could have getters to return the whole arrays instead?